src/pages/HomePage.tsx

```diff
@@ -7,61 +7,82 @@ import {
   CardContent,
   CircularProgress,
   Button,
+  Container,
+  Box,
 } from "@mui/material";
+import { createTheme, ThemeProvider } from "@mui/material/styles";
 import { Link } from "react-router-dom";
-
+import { Linkify } from "../utils/utilities";
 import { useAppDispatch } from "../redux/hooks";
 import { fetchPosts } from "../redux/post/post.actions";
 import { RootState } from "../redux/store";
+import { theme } from "../styles/theme";
+
+const defaultTheme = createTheme(theme);
 
 export const HomePage = (): JSX.Element => {
   const posts = useSelector((state: RootState) => state.posts);
-
+  const auth = useSelector((state: RootState) => state.auth);
   const dispatch = useAppDispatch();
+
   useEffect(() => {
     dispatch(fetchPosts());
   }, [dispatch]);
 
   return (
-    <Grid container direction="column" alignItems="center">
-      <Grid
-        container
-        flexDirection="row"
-        alignItems="center"
-        justifyContent="center"
-      >
-        <Typography variant="h4" sx={{ alignContent: "center" }}>
-          All posts
-        </Typography>
-        <Button
-          variant="contained"
-          href="/posts/new"
-          sx={{ width: 120, mx: 2 }}
+    <ThemeProvider theme={defaultTheme}>
```

Comment:
I've noticed that you use ThemeProvider in more than one spot, please change all of it so that you only use this once in you project. Conventionally this is set at the entry point of the app where your Redux wrapper also is.